File: tee_tool.py
from pathlib import Path
from check_file_permission import check_file_permission

def tee_file(file_path, content, config):
    if not check_file_permission(file_path, "write", config):
        raise PermissionError(f"Write access denied for {file_path}")
    file_path = Path(file_path).resolve()
    with open(file_path, "w", encoding="utf-8") as f:
        f.write(content)
    return content

File: ts-tools/tee_tool.ts

import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function teeFile(filePath: string, content: string, config: any): Promise<string> {
    return new Promise((resolve, reject) => {
        checkFilePermission(filePath, "write", config)
            .then(() => {
                const resolvedPath = path.resolve(filePath);
                fs.writeFile(resolvedPath, content, 'utf-8', (err) => {
                    if (err) {
                        reject(err);
                    } else {
                        resolve(content);
                    }
                });
            })
            .catch(reject);
    });
}
